cmux-tui/bindings/verify_github_environment_policy.py

```
@@ -0,0 +1,216 @@
+    if not REPOSITORY_PATTERN.fullmatch(repository):
+        raise EnvironmentPolicyError("GitHub repository must be OWNER/REPOSITORY")
+    return repository
+
+
+def _environment_url(
+    api_url: str,
+    repository: str,
+    environment: str,
+) -> str:
+    if not environment or environment.strip() != environment:
+        raise EnvironmentPolicyError("GitHub environment name must be non-empty")
```

Ruff (0.16.1)

[warning] Avoid specifying long messages outside the exception class

(TRY003)